fflas-ffpack/fflas/fflas_bounds.inl

```diff
@@ -115,11 +119,27 @@ namespace FFLAS {
     inline Givaro::Integer
     InfNorm (const size_t M, const size_t N, const Givaro::Integer* A, const size_t lda){
         Givaro::Integer max = 0;
-        for (size_t i=0; i<M; ++i)
-            for (size_t j=0; j<N; ++j) {
-                const Givaro::Integer & x(A[i*lda+j]);
-                if (Givaro::absCompare(x,max)>0) max = x;
-            }
+        std::vector<Givaro::Integer> vmax(M,0);
+        auto sp=SPLITTER(NUM_THREADS,FFLAS::CuttingStrategy::Row,FFLAS::StrategyParameter::Threads);
+        SYNCH_GROUP({
+              FORBLOCK1D(iter, M, sp,
+                          TASK(MODE(CONSTREFERENCE(A,max,vmax) ),
+                                {
+                                for(auto i=iter.begin(); i!=iter.end(); ++i)
+                                {
+                                    for (size_t j=0; j<N; ++j) {
+                                        const Givaro::Integer & x(A[i*lda+j]);
+                                        if (Givaro::absCompare(x,vmax[i])>0){ vmax[i] = x;}
```

Comment:
Updating vmax[i] in each parallel task creates contention on every operation. This code is likely not parallel at all.

Reply:
Yes not so correctly parallelized but timing showed a few speedup. I cannot see any easier way to find the local max value for each thread then search for the global max value outside the parallel region.